DigitalLearningSolutions.Data.Tests/Services/PasswordResetServiceTests.cs

```diff
@@ -269,6 +293,34 @@ public void GenerateAndScheduleDelegateWelcomeEmail_schedules_email()
                 .MustHaveHappened();
         }
 
+        [Test]
+        public void GenerateAndScheduleDelegateWelcomeEmail_throws_exception_with_incorrect_candidate_number()
+        {
+            // Given
+            var deliveryDate = new DateTime(2200, 1, 1);
+            const string emailAddress = "recipient@example.com";
+            const string addedByProcess = "some process";
+            const string candidateNumber = "A1";
+            var delegateUser = Builder<DelegateUser>.CreateNew()
+                .With(user => user.EmailAddress = emailAddress)
+                .Build();
+
+            A.CallTo(() => userService.GetDelegateUsersByEmailAddress(emailAddress))
+                .Returns(new List<DelegateUser> { delegateUser });
+
+            // Then
+            Assert.Throws<UserAccountNotFoundException>(
+                () =>
+                    passwordResetService.GenerateAndScheduleDelegateWelcomeEmail(
+                        emailAddress,
+                        candidateNumber,
+                        "example.com",
+                        deliveryDate,
+                        addedByProcess
+                    )
+            );
+        }
+
         public void SendWelcomeEmailsToDelegates_schedules_emails_to_delegates()

```

Comment:
I know this isn't yours but is it supposed to be a test? It's missing the annotation if so.